webrtc.html

```diff
@@ -8723,10 +8723,39 @@ <h3><dfn>RTCDataChannel</dfn></h3>
       <p>An <code><a>RTCDataChannel</a></code> object's <a>underlying data
       transport</a> may be torn down in a non-abrupt manner by running the
       <dfn id="data-transport-closing-procedure">closing procedure</dfn>. When
-      that happens the user agent MUST, unless the procedure was initiated by
-      the <code><a data-link-for="RTCDataChannel">close</a></code> method, queue a
-      task that sets the object's <a>[[\ReadyState]]</a> slot to <code>closing</code>.
-      This will eventually render the <a>data transport</a> <a>closed</a>.</p>
+      that happens the user agent MUST queue a task to run the following
+      steps:</p>
+      <ol>
+        <li>
+          <p>Let <var>channel</var> be the <code><a>RTCDataChannel</a></code>
+          object whose <a data-lt="data transport">transport</a> was
+          closed.</p>
+        </li>
+        <li>
+          <p>Unless the procedure was initiated by the <var>channel</var>'s
+          <code><a data-link-for="RTCDataChannel">close</a></code> method, set
+          <var>channel</var>'s <a>[[\ReadyState]]</a> slot to
+          <code>closing</code>.</p>
+        </li>
+        <li>
+          <p>Finish sending all currently pending messages of the
+          <var>channel</var>.</p>
```

Comment:
This can't be done directly from a queued task on the main thread; I think we need the whole "in parallel..." deal here.

Reply:
Would it be enough to mention that in the task? E.g.

> Finish sending all currently pending messages of the channel in parallel and wait for that procedure to complete.

Reply:
That may imply that the main thread is blocked while waiting, though.

Reply:
Mhh...

> Finish sending all currently pending messages of the channel in parallel and, in the task, wait for that procedure to complete.

Or do you have another suggestion?

Reply:
"In the task, wait for that procedure to complete" may still imply the main thread is blocked. We follow a model of "queued task = main thread", "in parallel = other threads". So I think it would be more consistent to structure it as:

- In parallel:
  - Finish sending all pending messages.
  - Follow the closing procedure.
  - Queue a task to render the channel closed.

Or in other words, there's one task that changes the state to "closing", then stuff happens in parallel, then there's another task that changes the state to "closed". Rather than one task that waits.